### PR TITLE
Refactor e2e tests

### LINKED ISSUE #262: Remove request body logging
https://github.com/substratusai/kubeai/blob/884aa62c81e2f5575a92c60a62f6e8734418a17b/internal/modelproxy/request.go#L131

This becomes especially problematic with binary requests (Speech to Text).

### PR DESCRIPTION
**Pull request by nstogner**
* Add central bash script to run e2e test cases
* Separate e2e tests out into separate cases
* Add python openai client test case
* Fix #262

### CODE REVIEW

File: skaffold.yaml [hunk 1]
```diff
@@ -5,20 +5,39 @@ metadata:
...
     push: false
 deploy:
   helm:
     releases:
     - name: kubeai
-      chartPath: charts/kubeai
-      valuesFiles:
-      - charts/kubeai/values.yaml
-      setValueTemplates:
-        image.tag: "{{.DIGEST_HEX}}"
```
Comment (samos123) [inline skaffold.yaml:18]: i think i needed this to ensure it uses the current revision

Comment (nstogner) [reply skaffold.yaml:18]: `skaffold render` shows that it mutates the image tag itself: `image: substratusai/kubeai:helm-chart-kubeai-0.6.0-4-gcdeaea1`

**Comment by nstogner** [comment]
End-to-end tests run in parallel (in sequence after unit and integration tests - if those fail, theres not use in tying up resources to run e2e tests):
Screenshot 2024-10-07 at 8 25 35 PM

**Comment by nstogner** [comment]
Should be good to merge if all looks OK

**Comment by samos123** [comment]
Nice! LGTM.

I think I rather run e2e and unit /integration test at the same time so we don't have to wait an additional 4 minutes. I can tackle that in a follow up, lets merge this first.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -8,38 +8,42 @@ on:
...
 
-      - name: Run unit & integration tests
-        run: make test
-  e2e-quickstart:
+      - name: Run integration tests
+        run: make test-integration
+
+  e2e:
     runs-on: ubuntu-latest
+    needs: unit-and-integration # No use in running e2e tests if integration tests fail.
```
Comment (samos123) [inline .github/workflows/tests.yml:25]: I would prefer to remove this needs

Comment (samos123) [reply .github/workflows/tests.yml:25]: if we start running out of resources, then we can consider adding it back.

**Review by samos123 — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (nstogner) [reply .github/workflows/tests.yml]: done

**Pull request merged**